### ISSUE TITLE
Harvest problem

### ISSUE THREAD

**Issue by HenryLoenwind** [HenryLoenwind/EnderIOAddons#15]
GlimmermanFX4:
I was seriously loving this mod and the idea of the agricraft farming station until i started trying to actually farm the capacitor seeds.  I have made sure they are set up exactly as you have described and on their mouse over tool tip it even says they are fertile and can grow.  The issue is when they are leveled up (all 10/10/10) they only give 1 crop ever and then never produce after that.  I have had a farming station with these active for days now along with an autonomous activator and watering can to make them grow and have never gotten more than what the initial planting yielded.  Is this by design or is this a bug? I'm seriously hoping this is some kind of bug or programming over site because if this is indeed by design it would most likely make this mod unplayable in my case at least. Everything in this latest update seems to depend solely on those plants and the harvesting of them.

TsumiHokiro:
I'm having the same problem as Glimmerman. I have planted the Capacitor seeds and have upgraded them till 10/10/10. While I was doing it, the first crop or first few ones I was able to harvest the crop, but as it went on, the harvest was diminishing to the point that, when I got to the perfect seeds, I wasn't able to harvest anything anymore. At first I thought that maybe it wasn't 100% sure that whenever you harvest it it yields fruits even though it accused being mature, however after several tries, and after planting several others, none of them, neither the original nor the other ones, were giving me fruits. As Glimmerman said, I wonder if this is a bug or something?.

If it helps, I'm running the game on a local server, and once I've restarted the server it does happens to give fruits, on the first few times it, but as I attempt to harvest it further on, it does not give anything at all, from none of the plants that I have planted (3 in total).

GlimmermanFX4:
Basically the same issue as TsumiHokiro explained below.  I may get one or two crops off the plants but after that they completely stop producing.  I'll dig them up and replant them and still they won't produce.

GlimmermanFX4:
They do grow and wailia shows them as mature when i mouse over then and so does the graphic of the plant but when harvested they will not drop anything and simply disappear as normal and start to grow again.

**Comment by HenryLoenwind** [comment]
Can you try with this version:

http://loenwind.info/eioa/EnderIOAddons-1.7.10-2.3.0.421_beta-0.9.9.037_beta.jar

And report me what it writes to the (server) logfile when a plant is harvested? It should look like this:

```
[08:21:52] [Server thread/INFO] [enderioaddons]: Starting harvest for a capacitor plant with gain 6
[08:21:52] [Server thread/INFO] [enderioaddons]: Adding low level fruit: 1xitem.itemMachineParts@23
[08:21:52] [Server thread/INFO] [enderioaddons]: Adding low level fruit: 1xitem.itemMachineParts@23
[08:21:52] [Server thread/INFO] [enderioaddons]: Adding high level fruit: 1xitem.itemMachineParts@24
[08:21:52] [Server thread/INFO] [enderioaddons]: Harvest finished with 3 fruits: [1xitem.itemMachineParts@23, 1xitem.itemMachineParts@23, 1xitem.itemMachineParts@24]
```

**Comment by lilrichy** [comment]
Hi I am the server owner for the server that GlimmermanFX4 plays on we ran the test that you asked us to. This version seems to be working correctly! This is the log:

[03:55:33] [Server thread/DEBUG] [FML/]: Gathering id map for writing to world save MEGAWorld
[03:55:37] [Server thread/INFO] [enderioaddons/]: Starting harvest for a capacitor plant with gain 10
[03:55:37] [Server thread/INFO] [enderioaddons/]: Adding low level fruit: 1xitem.itemMachineParts@23
[03:55:37] [Server thread/INFO] [enderioaddons/]: Adding low level fruit: 1xitem.itemMachineParts@23
[03:55:37] [Server thread/INFO] [enderioaddons/]: Adding high level fruit: 1xitem.itemMachineParts@23
[03:55:37] [Server thread/INFO] [enderioaddons/]: Adding high level fruit: 1xitem.itemMachineParts@23
[03:55:37] [Server thread/INFO] [enderioaddons/]: Harvest finished with 4 fruits: [1xitem.itemMachineParts@23, 1xitem.itemMachineParts@23, 1xitem.itemMachineParts@23, 1xitem.itemMachineParts@23]
[03:55:39] [Server thread/INFO] [enderioaddons/]: Starting harvest for a capacitor plant with gain 10
[03:55:39] [Server thread/INFO] [enderioaddons/]: Adding low level fruit: 1xitem.itemMachineParts@23
[03:55:39] [Server thread/INFO] [enderioaddons/]: Adding low level fruit: 1xitem.itemMachineParts@23
[03:55:39] [Server thread/INFO] [enderioaddons/]: Adding high level fruit: 1xitem.itemMachineParts@23
[03:55:39] [Server thread/INFO] [enderioaddons/]: Adding high level fruit: 1xitem.itemMachineParts@23
[03:55:39] [Server thread/INFO] [enderioaddons/]: Harvest finished with 4 fruits: [1xitem.itemMachineParts@23, 1xitem.itemMachineParts@23, 1xitem.itemMachineParts@23, 1xitem.itemMachineParts@23]
[03:55:40] [Server thread/INFO] [enderioaddons/]: Starting harvest for a capacitor plant with gain 10
[03:55:40] [Server thread/INFO] [enderioaddons/]: Adding low level fruit: 1xitem.itemMachineParts@23
[03:55:40] [Server thread/INFO] [enderioaddons/]: Adding low level fruit: 1xitem.itemMachineParts@23
[03:55:40] [Server thread/INFO] [enderioaddons/]: Adding high level fruit: 1xitem.itemMachineParts@23
[03:55:40] [Server thread/INFO] [enderioaddons/]: Adding high level fruit: 1xitem.itemMachineParts@23
[03:55:40] [Server thread/INFO] [enderioaddons/]: Harvest finished with 4 fruits: [1xitem.itemMachineParts@23, 1xitem.itemMachineParts@23, 1xitem.itemMachineParts@23, 1xitem.itemMachineParts@23]

**Comment by lilrichy** [comment]
Also the version my server was running that had the problem was: EnderIOAddons-1.7.10-2.3.0.420_beta-0.9.7.031_beta
The server will be updating to the EnderIOAddons-1.7.10-2.3.0.421_beta-0.9.9.036_beta version once curse pushes my modpack update.

**Comment by HenryLoenwind** [comment]
Ok, as the only difference between the latest release and the version I gave you is the debug output, I'll assume whatever the cause was, it is fixed in the latest release.
